llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -4589,6 +4589,18 @@ def : Pat<(truncstorei16 GPR64:$Rt, (am_unscaled16 GPR64sp:$Rn, simm9:$offset)),
 def : Pat<(truncstorei8 GPR64:$Rt, (am_unscaled8 GPR64sp:$Rn, simm9:$offset)),
   (STURBBi (EXTRACT_SUBREG GPR64:$Rt, sub_32), GPR64sp:$Rn, simm9:$offset)>;
 
+// v1i64 -> bsub truncating stores
+// Supporting pattern lower f32/64 -> v8i8
+def : Pat<(v8i8 (vector_insert (v8i8 (undef)), (i32 FPR32:$src), 0)),
+          (INSERT_SUBREG (v8i8 (IMPLICIT_DEF)), FPR32:$src, ssub)>;
+def : Pat<(v8i8 (vector_insert (v8i8 (undef)), (i64 FPR64:$src), 0)),
+          (v8i8 (EXTRACT_SUBREG (INSERT_SUBREG (v16i8 (IMPLICIT_DEF)), FPR64:$src, dsub), dsub))>;
+// Lower v1i64 -> v1i8 truncstore to bsub store
+def : Pat<(truncstorevi8 v1i64:$VT, (am_unscaled8 GPR64sp:$Rn, simm9:$offset)),
+          (STURBi (vi8 (EXTRACT_SUBREG v1i64:$VT, bsub)), GPR64sp:$Rn, simm9:$offset)>;
+def : Pat<(truncstorevi8 v1i64:$VT, (am_indexed8 GPR64sp:$Rn, uimm12s4:$offset)),
+          (STRBui (vi8 (EXTRACT_SUBREG v1i64:$VT, bsub)), GPR64sp:$Rn, uimm12s4:$offset)>;
```

Comment:
I can't figure out a way to write this these lowerings that does not make use of `vi8`. I've tried `i8 FRP8` rather than `vi8` but that is not syntactically valid.

Reply:
I hoped we could use something like `(i8 (COPY_TO_REGCLASS v1i64:$VT, FPR8))` but the downside to that is we then need to implement support to copy across the register classes within `copyPhysReg` which is generally frowned upon.  That alone might not be terminal but because the register classes do not match you do end up with redundant copies where the source and destination are the same.

Reply:
I find it weird that `STRBui` is defined in such a way as to require an MVT that is not legal and whilst you kind of fix this by introducing a new "legal" type, it seems equally weird to need a new MVT for a single instruction.

Reply:
Well for every other type where we'd want to do something like this `i16`, `i32`, etc, there already was a legal FPR type we could use for the `EXTRACT_SUBREG`. Really by adding `vi8` we're just handling this the same as every other FPR type, it's just there wasn't a legal FP8 type we could use already.

Reply:
Note: There are a bunch of other target-specific MVTs next to where I placed `vi8` (including some for AArch64), so I thought it was a reasonable, and fairly unobtrusive solution.

Reply:
Sure, I get that and I'm not against the solution.  I'm just applying caution because the use case here is so minimal and if there's a path to avoid it then I'd rather take it.  That said, the available options are running.